launcher/modManager/cmodlist.cpp

```diff
@@ -208,11 +217,29 @@ CModEntry CModList::getMod(QString modname) const
 	}
 	else
 	{
-		if(conf.canConvert<QVariantMap>())
+		if(!conf.toMap().isEmpty())
+		{
 			settings = conf.toMap();
+			if(settings.value("active").isNull())
+				settings["active"] = true; // default
+		}
 		else
 			settings.insert("active", conf);
```

Comment:
not directly related to your PR, but this line is suspicious. `settings["active"]` is treated everywhere as a boolean, but here whatever is in `conf` is inserted. Are there guarantees that it can only be a boolean? I'd rather change `else` to `else if(conf.canConvert<bool>())`, and in a new `else` branch at least print some error message.